app/src/main/java/com/example/parkingsystemkotlin/util/ReservationVerify.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.parkingsystemkotlin.util
+
+import com.example.parkingsystemkotlin.database.ParkingSpaceReservationDB
+import com.example.parkingsystemkotlin.entity.Reservation
+
+class ReservationVerify(var database: ParkingSpaceReservationDB) {
+
+    fun canBeReserved(reservation: Reservation): Boolean {
+        val parkingSpace = reservation.parkingSpace
+        if (database.hashReservation.containsKey(parkingSpace)) {
+            val reservations: MutableList<Reservation>? = database.hashReservation[parkingSpace]
+            reservations?.let {
+                for (r: Reservation in reservations) {
```

Comment:
Instead to do this, you can use reservations.forEach { ... }

Reply:
Done!